src/utils/api.js

```diff
@@ -1,8 +1,31 @@
-export const getData = async url => {
-  const response = await fetch(url);
-  if (!response.ok) {
-    const error = await response.json();
-    throw new Error(error.message);
+export const fetchData = async (path, method, data = null) => {
+  let options;
+  switch (method) {
+    case 'DELETE':
+      options = { method };
+      break
+    case 'GET':
+      options = data;
+      break
+    default:
+      options = {
+        method,
+        body: JSON.stringify(data),
+        headers: {
+          'Content-type': 'application/json'
+        }
+      };
+  };
```

Comment:
I LOVE the switch statement here! I didn't use one in my prior fetchData function that took in multiple methods and this makes it much more readable. Nice work.